library/pk.c

```diff
@@ -197,7 +197,7 @@ int mbedtls_pk_setup_opaque(mbedtls_pk_context *ctx,
 }
 #endif /* MBEDTLS_USE_PSA_CRYPTO */
 
-#if defined(MBEDTLS_PK_USE_PSA_EC_DATA)
+#if defined(MBEDTLS_PK_USE_PSA_EC_DATA) && defined(MBEDTLS_ECP_LIGHT)
```

Comment:
The guard is correct, but taking a step back, why do we still need this function in the library? It looks like it's only used in `test_suite_pk.function`, so perhaps it should live there instead? Or even better, perhaps the code using it can be refactored to no longer need it.

Reply:
Yes, absolutely. It turned out to be extremely easy to make it work without ECP_LIGHT. Thanks for the suggestion!